### ISSUE TITLE
[Bug]: Gifs no longer create transparent background

### ISSUE THREAD

**Issue by JohannesWal** [matplotlib/matplotlib#27173]
### Bug summary

Matplotlib is no longer able to create gifs with a transparent background.

### Code for reproduction

```python
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.animation import FuncAnimation

fig, ax = plt.subplots()
xdata, ydata = [], []
ln, = ax.plot([], [], 'ro')

def init():
    ax.set_xlim(0, 2*np.pi)
    ax.set_ylim(-1, 1)
    return ln,

def update(frame):
    xdata.append(frame)
    ydata.append(np.sin(frame))
    ln.set_data(xdata, ydata)
    return ln,

ani = FuncAnimation(fig, update, frames=np.linspace(0, 2*np.pi, 128),
                    init_func=init, blit=True)


savefig_kwargs = {
    "facecolor": "none",
}
ani.save("test.gif", savefig_kwargs=savefig_kwargs)
```

### Actual outcome

With Matplotlib version 3.8.0 (and also 3.7.3) installed this is the ouput:
![3 8 0](https://github.com/matplotlib/matplotlib/assets/148774400/558b06cb-ba89-446d-97a3-b5e8aa2d7b1f)

A gif-file with a white background

### Expected outcome

In 3.6.3 and before the file had a transparent background

![3 6 3](https://github.com/matplotlib/matplotlib/assets/148774400/daf64674-e396-41e1-8444-084b40ff07dd)

### Additional information

I tried many different setups, different values for facecolor (e.g. (1,1,1,1)), set "transparent": True and googeling quite a lot - old solutions on github won't work.

Like in the expected outcome stated, this worked in the past (until Matplotlib 3.6.3).

There either seems to be an error in the fix for another issue in #21831 - or this is now the new correct behavior, but it seems like the transparency should still work for gifs, as this was an issue with different formats.

### Operating system

Windows 11

### Matplotlib Version

3.8.0

### Matplotlib Backend

TkAgg

### Python version

Python 3.12.0

### Installation

pip

**Comment by cgadal** [comment]
It seems it comes from:

https://github.com/matplotlib/matplotlib/blob/3ecb14885c4803ea034ddb4b778a02d33366882f/lib/matplotlib/animation.py#L1080-L1084

where `pre_composite_to_white` matches `none` to a white color. 

Is this intended behaviour @tacaswell (I think you added this ?)

**Comment by Wuzuw** [comment]
@cgadal @JohannesWal @tacaswell 

Me and two fellow students have been looking into this issue as part of a project.
So far we have found that the fix you mentioned takes alpha and therefore transparency out of the equation. As far as we understand it simulates transparency by scaling the color and making it appear "faded" over a white background. But actually it transforms the rgba value to a rgb value (no transparency).

If you hardcode a rgba value then the previous problem returns (the blocky looks). However only if you don't use ffmpeg and only the pillowwriter.

We think this could be solved by checking which writer is used and only allowing "true" transparency when using ffmpeg and using the previous fix without it.

**Comment by trochitv** [comment]
I have a similar issue with transparent .mov files: https://discourse.matplotlib.org/t/background-transparency-lost-with-ffmpeg/24189
Seems sad that I have to rollback to version 3.6.3 to keep on creating these.

**Comment by tacaswell** [comment]
Sorry I missed this and the PR when it came in.

I agree that my fix for #19040 was too through.  #27406 is on the right track but needs some more work.

**Comment by QuLogic** [comment]
So I had a look into this, and it appears that a) we may want to allow more than just GIF, b) it should be allowed for all writers, and c) GIF doesn't have true transparency. The latter point is a bit of an annoyance, as Pillow handles this differently than ffmpeg/imagemagick.

Pillow appears to make anything that's partially transparent into fully opaque pixels:
![test-pillow](https://github.com/user-attachments/assets/c1d97f5c-803a-457c-979a-8232d1d2c7fd)
but ffmpeg and imagemagick make them fully transparent:
![test-ffmpeg](https://github.com/user-attachments/assets/bdc6f0e5-5b90-4f0f-8734-fd28de1be7bb)
Neither looks great for text, but I'm not sure whether we should attempt to normalize those.
